src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.hidden) {
+      return;
+    }
+
+    if (animal instanceof Carnivore) {
+      return;
+    }
+
+    animal.health -= 50;
+
+    if (animal.health === 0) {
```

Comment:
The condition `if (animal.health === 0)` should be `if (animal.health <= 0)` to ensure that animals with negative health are also removed from the `alive` array.